var/spack/repos/builtin/packages/umt/package.py

```diff
@@ -0,0 +1,196 @@
+        return sys_type
+
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip("1234567890")
+        return "{0}-{1}-{2}@{3}.cmake".format(
+            hostname,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version,
+        )
+
+    def cmake_args(self):
+        options = []
+        return options
+
+    def _get_host_config_path(self, spec):
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        host_config_path = "{0}-{1}-{2}-{3}.cmake".format(
+            socket.gethostname(), sys_type, spec.compiler, spec.dag_hash()
+        )
+        dest_dir = spec.prefix
+        host_config_path = os.path.abspath(join_path(dest_dir, host_config_path))
+
+        return host_config_path
+
+    # Override the parent class function until its fixed in spack.  (Greg Becker and Chris White are aware of bug)
+    def flag_handler(self, name, flags):
+        if name in ("cflags", "cxxflags", "cppflags", "fflags", "ldflags"):
+            return (None, None, None)  # handled in the cmake cache
+        return (flags, None, None)
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+
+        if "umpire" in spec:
+            if "+rocm" in spec["umpire"]:
+                spec.compiler_flags["cxxflags"].append("-I{0}/include".format(spec["hip"].prefix))
+
+        #######################
+        # Note - call the super class AFTER changing any flags, as the super class
+        # adds the cflags, cxxflags, fflags, ldflags, etc, to the cache entries list.
+        # If you try adding any of these yourself you will end up with duplicates.
+        # - black27
+        entries = super(Umt, self).initconfig_compiler_entries()
+
+        if "+openmp" in spec:
+            if spec.satisfies("%xl"):
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", "-qsmp=omp"))
+
+                if "+openmp" in spec:
+                    entries.append(cmake_cache_string("TETON_OpenMP_Offload_Fortran_FLAGS", "-qoffload -qtgtarch=sm_70"))
+            else:
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", "-fopenmp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", "-fopenmp"))
```

Comment:
```suggestion
            entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", self.compiler.openmp_flag))
            entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", self.compiler.openmp_flag))
            entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", self.compiler.openmp_flag))
                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", self.compiler.openmp_flag))
```

Reply:
This is a bit tricky for XL.  XL has two different openmp flags for OpenMP, and enabling OpenMP target offload kernels.
"-qsmp=omp" enables OpenMP, but not 'target' constructs.
"-qsmp=omp -qoffload" enables OpenMP and target constructs.

But I'll probably deprecate support for both of these and just allow the latter line or no line.  I need to double check if self.compiler.openmp_flags provides the full "-qsmp=omp -qoffload" on IBM platforms.